### DISCUSSION TITLE
Two auth methods

### DISCUSSION THREAD

**Discussion by zchira** [poem-web/poem#627] [Q&A] [answered]
Is it possible to allow calling some endpoint if only one of two Security Schemas are satisfied?

OpenApi spec says:

> Security Requirement Object
> 
> Lists the required security schemes to execute this operation. The name used for each property MUST correspond to a security scheme declared in the [Security Schemes](https://swagger.io/specification/#components-security-schemes) under the [Components Object](https://swagger.io/specification/#components-object).
> 
> Security Requirement Objects that contain multiple schemes require that all schemes MUST be satisfied for a request to be authorized. This enables support for scenarios where multiple query parameters or HTTP headers are required to convey security information.
> 
> When a list of Security Requirement Objects is defined on the [OpenAPI Object](https://swagger.io/specification/#oas-object) or [Operation Object](https://swagger.io/specification/#operation-object), only one of the Security Requirement Objects in the list needs to be satisfied to authorize the request.

The idea is to create API which can be used from the Web Interface or as an API.
In the first case the endpoint `/auth` would return `set-cookie` in the header etc...
In the second case the endpoint `/auth-api` would return api-key which will be used in further requests as a part of header (X-Token: xxx, or Authorization: Bearer xxx)

**Comment by sunli829** [comment, marked as answer]
It has been added in `v3.0.2`

https://github.com/poem-web/poem/blob/a7aca56ca9f11ff9417d5652dea879fcb8af2fbe/poem-openapi/tests/security_scheme.rs#L544
